test/test_functional_tensor.py

```diff
@@ -552,24 +552,25 @@ def _test_affine_translations(self, tensor, pil_img, scripted_affine):
     def _test_affine_all_ops(self, tensor, pil_img, scripted_affine):
         # 4) Test rotation + translation + scale + share
         test_configs = [
-            (45, [5, 6], 1.0, [0.0, 0.0]),
-            (33, (5, -4), 1.0, [0.0, 0.0]),
-            (45, [-5, 4], 1.2, [0.0, 0.0]),
-            (33, (-4, -8), 2.0, [0.0, 0.0]),
-            (85, (10, -10), 0.7, [0.0, 0.0]),
-            (0, [0, 0], 1.0, [35.0, ]),
-            (-25, [0, 0], 1.2, [0.0, 15.0]),
-            (-45, [-10, 0], 0.7, [2.0, 5.0]),
-            (-45, [-10, -10], 1.2, [4.0, 5.0]),
-            (-90, [0, 0], 1.0, [0.0, 0.0]),
+            (45.5, [5, 6], 1.0, [0.0, 0.0], None),
+            (33, (5, -4), 1.0, [0.0, 0.0], [0, 0, 0]),
+            (45, [-5, 4], 1.2, [0.0, 0.0], (1.0, 2.0, 3.0)),
```

Comment:
I changed here, but probably this will fail for PIL images as you said that for int PIL images fill value should not be float...
I can revert this commit.

Reply:
Yeah this will probably fail for PIL images, maybe change the f_pil line?

Reply:
> @voldemortX thanks a lot for the update ! Looks good now.
> Let me do another pass a bit later and probably it will be good to merge...

Thanks a lot for your guidance and reviews! This pr is much harder than I initially thought, I couldn't have done all this without your help.